### ISSUE TITLE
Cache-Control problems

### ISSUE THREAD

**Issue by tsubik** [kwhitley/apicache#74]
Hey @kwhitley I was trying to use this library, but what I see it sets "Cache-Control" to chosen cache duration. I was intending to spare the server heavy database tasks and the cache is controlled by the server, meaning it could be cleared anytime. Setting "Cache-Control" with fixed max-age prevents some clients from hitting my endpoints, even when the cache was cleared by the server. Any workarounds? I believe Cache-Control shouldn't be set in this library or it should be an option.

**Comment by tsubik** [comment]
All right, I can add another middleware after cache middleware to set Cache-Control to other value, but still I think that is a mistake to set Cache-Control to max-age.

**Comment by dotob** [comment]
Hi i want to second tsubik's request to not set this header. or make it optional. or is there a reasonable ides behind it that I do not understand?

From this apart I really like apicache! keep on the good work.

**Comment by Balgorn** [comment]
So basically was dealing with the same problem. What I did right now to get around this issue, is to set the duration in the middleware to a very high value (like 1 month). Afterwards I just add a middleware like this:

**Comment by Balgorn** [comment]
@tsubik So basically was dealing with the same problem. What I did right now to get around this issue is to set the duration in the middleware to a very high value (like 1 month). Afterwards I just add a middleware like this:
```
router.use((req, res, next) => {
	//needed for clients not to get cached content for ages and never retry
	res.header('cache-control', 'no-cache');
	next();
});
```

**Comment by kwhitley** [comment]
@tsubik @dotob @Balgorn This will be resolved in https://github.com/kwhitley/apicache/pull/97.  I left the default behavior in to prevent breakage in existing deployed apps, but added the ability to override headers via global (or local middleware) options:

```js
let cache = apicache.options({ headers: { 'cache-control': 'no-cache' } })
```

That work?

**Comment by Balgorn** [comment]
That should do it. If you can still set the maximum duration for your local Cache as before.

**Comment by kwhitley** [comment]
@tsubik @Balgorn @dotob - Ability to override headers via headers option (above) has been implemented in v0.9.0 and test coverage included to verify.  Feel free to test it out!
